[BUG] Plugin crashes when opening big files

# Description
When opening a file, the plugin tries to get the document's text as a background process. When the user opens a very big file, this throws an `AssertionError`, stacktrace below.

# Environment
- OS: Elementary OS 4.15.0-58-generic
- IDE: IntelliJ 2019.3.4
- Plugin version: 2.1.0
- Java version: 11

# Stacktrace
```
java.lang.AssertionError: File text mismatch: tree.length=200021; psi.length=0; this=PsiFile(plain text):test.txt; vp=com.intellij.psi.SingleRootFileViewProvider{myVirtualFile=file:///home/stijn/Documents/Project AD2/01903503/files/test.txt, content=VirtualFileContent{size=89441718}}
	at com.intellij.psi.impl.source.PsiFileImpl.getText(PsiFileImpl.java:288)
	at java.base/java.util.Optional.map(Optional.java:265)
	at io.github.thepieterdc.dodona.plugin.exercise.identification.impl.IdentificationServiceImpl.identify(IdentificationServiceImpl.java:51)
	at io.github.thepieterdc.dodona.plugin.events.ExerciseOpenedBroadcaster.lambda$selectionChanged$0(ExerciseOpenedBroadcaster.java:59)
	at java.base/java.util.Optional.flatMap(Optional.java:294)
	at io.github.thepieterdc.dodona.plugin.events.ExerciseOpenedBroadcaster.selectionChanged(ExerciseOpenedBroadcaster.java:59)
	at jdk.internal.reflect.GeneratedMethodAccessor85.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.intellij.util.messages.impl.MessageBusImpl.invokeListener(MessageBusImpl.java:513)
	at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:142)
	at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:438)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:398)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:388)
	at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:372)
	at com.intellij.util.messages.impl.MessageBusImpl.lambda$createTopicHandler$1(MessageBusImpl.java:241)
	at com.sun.proxy.$Proxy98.selectionChanged(Unknown Source)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.lambda$fireSelectionChanged$21(FileEditorManagerImpl.java:1576)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl$6$1.run(FileEditorManagerImpl.java:1019)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.lambda$doWhenFocusSettlesDown$3(FocusManagerImpl.java:168)
	at com.intellij.util.ui.UIUtil.invokeLaterIfNeeded(UIUtil.java:2418)
	at com.intellij.ide.IdeEventQueue.ifFocusEventsInTheQueue(IdeEventQueue.java:177)
	at com.intellij.ide.IdeEventQueue.executeWhenAllFocusEventsLeftTheQueue(IdeEventQueue.java:129)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:159)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:165)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.lambda$null$2(FocusManagerImpl.java:173)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:309)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:441)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:424)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:407)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:908)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:781)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:424)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:698)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:423)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

--- a/build.gradle
+++ b/build.gradle
@@ -58,6 +58,7 @@ jacocoTestReport {
 patchPluginXml {
     changeNotes """
       <ul>
+        <li>[BUGFIX] Opening large files will no longer show an error message.</li>
       </ul>
       """
 

--- a/src/main/java/io/github/thepieterdc/dodona/plugin/exercise/identification/impl/IdentificationServiceImpl.java
+++ b/src/main/java/io/github/thepieterdc/dodona/plugin/exercise/identification/impl/IdentificationServiceImpl.java
@@ -15,6 +15,7 @@ import com.intellij.psi.PsiElement;
 import com.intellij.psi.PsiManager;
 import io.github.thepieterdc.dodona.plugin.exercise.Identification;
 import io.github.thepieterdc.dodona.plugin.exercise.identification.IdentificationService;
+import io.github.thepieterdc.dodona.plugin.notifications.ErrorReporter;
 import io.github.thepieterdc.dodona.resources.Course;
 import io.github.thepieterdc.dodona.resources.Series;
 import io.github.thepieterdc.dodona.resources.activities.Activity;
@@ -46,9 +47,15 @@ public class IdentificationServiceImpl implements IdentificationService {
 	@Override
 	public Optional<Identification> identify(final Project project,
 	                                         final VirtualFile file) {
-		return Optional.of(PsiManager.getInstance(project))
-			.map(psiMgr -> psiMgr.findFile(file))
-			.map(PsiElement::getText)
-			.flatMap(this::identify);
+		try {
+			return Optional.of(PsiManager.getInstance(project))
+				.map(psiMgr -> psiMgr.findFile(file))
+				.map(PsiElement::getText)
+				.flatMap(this::identify);
+		} catch (final Throwable t) {
+			// Handle files that are too large.
+			ErrorReporter.report(t);
+			return Optional.empty();
+		}
 	}
 }

